metadata-service/factories/src/main/java/com/linkedin/gms/factory/auth/AuthorizerChainFactory.java

```diff
@@ -65,53 +80,67 @@ protected AuthorizerChain getInstance() {
     return new AuthorizerChain(authorizers, dataHubAuthorizer);
   }
 
-  private AuthorizerContext initAuthorizerContext() {
-    final ResourceSpecResolver resolver = new DefaultResourceSpecResolver(systemAuthentication, entityClient);
-    return new AuthorizerContext(resolver);
+  private ResourceSpecResolver initResolver() {
+    return new DefaultResourceSpecResolver(systemAuthentication, entityClient);
   }
 
-  private List<Authorizer> initCustomAuthorizers(AuthorizerContext ctx) {
+  private List<Authorizer> initCustomAuthorizers(ResourceSpecResolver resolver) {
     final List<Authorizer> customAuthorizers = new ArrayList<>();
 
-    if (this.configurationProvider.getAuthorization().getAuthorizers() != null) {
-
-      final List<AuthorizerConfiguration> authorizerConfigurations =
-          this.configurationProvider.getAuthorization().getAuthorizers();
-
-      for (AuthorizerConfiguration authorizer : authorizerConfigurations) {
-        final String type = authorizer.getType();
-        // continue if authorizer is not enabled
-        if (!authorizer.isEnabled()) {
-          log.info(String.format("Authorizer %s is not enabled", type));
-          continue;
-        }
-
-        final Map<String, Object> configs =
-            authorizer.getConfigs() != null ? authorizer.getConfigs() : Collections.emptyMap();
-
-        log.debug(String.format("Found configs for notification sink of type %s: %s ", type, configs));
-
-        // Instantiate the Authorizer
-        Class<? extends Authorizer> clazz = null;
-        try {
-          clazz = (Class<? extends Authorizer>) Class.forName(type);
-        } catch (ClassNotFoundException e) {
-          throw new RuntimeException(
-              String.format("Failed to find custom Authorizer class with name %s on the classpath.", type));
-        }
-
-        // Else construct an instance of the class, each class should have an empty constructor.
-        try {
-          final Authorizer authorizerInstance = clazz.newInstance();
-          authorizerInstance.init(configs, ctx);
-          customAuthorizers.add(authorizerInstance);
-          log.info(String.format("Authorizer %s is initialized", type));
-        } catch (Exception e) {
-          throw new RuntimeException(
-              String.format("Failed to instantiate custom Authorizer with class name %s", clazz.getCanonicalName()), e);
-        }
-      }
-    }
+    Path pluginBaseDirectory = Paths.get(configurationProvider.getDatahub().getPlugin().getAuth().getPath());
```

Comment:
Question -> Why is this configuration under "datahub"? Why not just under plugins?

Reply:
In application.yml the entityRegistry and retention was under datahub -> plugin, so I just created one entry for auth